web-crypto/sign-verify/rsa-pss.js

```diff
@@ -0,0 +1,95 @@
+(function() {
+
+  /*
+  Store the calculated signature here, so we can verify it later.
+  */
+  let signature;
+
+  /*
+  Fetch the contents of the "message" textbox, and encode it
+  in a form we can use for sign operation.
+  */
+  function getMessageEncoding() {
+    const messageBox = document.querySelector(".rsa-pss #message");
+    let message = messageBox.value;
+    let enc = new TextEncoder();
+    return enc.encode(message);
+  }
+
+  /*
+  Get the encoded message-to-sign, sign it and display a representation
+  of the first part of it in the "signature" element.
+  */
+  async function signMessage(privateKey) {
+    const signatureValue = document.querySelector(".rsa-pss .signature-value");
+    signatureValue.classList.remove("valid");
+    signatureValue.classList.remove("invalid");
+
+    let encoded = getMessageEncoding();
+    signature = await window.crypto.subtle.sign(
+      {
+        name: "RSA-PSS",
+        saltLength: 32,
+      },
+      privateKey,
+      encoded
+    );
+
+    let buffer = new Uint8Array(signature, 0, 10);
+    signatureValue.textContent = buffer;
+  }
+
+  /*
+  Fetch the encoded message-to-sign and verify it against the stored signature.
+  * If it checks out, set the "valid" class on the signature.
+  * Otherwise set the "invalid" class.
+  */
+  async function verifyMessage(publicKey) {
+    const signatureValue = document.querySelector(".rsa-pss .signature-value");
+    signatureValue.classList.remove("valid");
+    signatureValue.classList.remove("invalid");
+
+    let encoded = getMessageEncoding();
+    let result = await window.crypto.subtle.verify(
```

Comment:
The current documentation for `subtle.crypto.verify()` only mentions `HMAC`, `RSASSA-PKCS1-v1_5` and `ECDSA` as options. It also lists the algorithm as a `DOMString`, not an object. I'm not sure if that's just a simple tweak or if there is intended to be an overhaul of the function documentation as well.

Reply:
Yes, those are errors in the docs:

* RSA-PSS is also supported (https://www.w3.org/TR/WebCryptoAPI/#rsa-pss).

* Wherever `algorithm` is an object I want to have a separate page describing the object (i.e. `RsaPssParams`, `EcdsaParams`, that's what I'm hoping to do in https://github.com/mdn/sprints/issues/667.